src/api/check.rs

```diff
@@ -1090,42 +1025,59 @@ impl ResponseWithContext {
         }
 
         self.response.matches.append(&mut other.response.matches);
-        self.text.push_str(other.text.as_str());
-        self.text_length += other.text_length;
+
+        self.text = Cow::Owned(self.text.into_owned() + &other.text);
```

Comment:
Probably "nicer" to use https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.to_mut instead, though this eventually does the same

Reply:
I can change that yeah - never properly used `Cow` before to be honest, good to learn all these